cryptolib.c

```diff
@@ -36,6 +36,10 @@
 #include <config.h>
 #include <string.h>
 
+#if defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER > 0x3040000fL)
```

Comment:
The final ) is wrong.     All the other version checks seem to use 0x3050000fL rather than 0x3040000fL.    Is that correct?

In general I'd prefer a little abstraction.    It seems there is an API change in OpenSSL 0x10100000L and LibreSSL 0x3050000fL that covers all these changes.    What about inventing some cpp symbol for that and use that consistently?